src/Tool/RequestFactory.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace League\OAuth2\Client\Tool;
+
+use GuzzleHttp\Psr7\Request;
+
+// For history see https://github.com/guzzle/guzzle/pull/1101
+class RequestFactory
+{
+    /**
+     * @param  null|string $method HTTP method for the request.
+     * @param  null|string $uri URI for the request.
+     * @param  array  $headers Headers for the message.
+     * @param  string|resource|StreamInterface $body Message body.
+     * @param  string $protocolVersion HTTP protocol version.
+     * @return Request
+     */
+    public function getRequest(
+        $method,
+        $uri,
+        array $headers = [],
+        $body = null,
+        $protocolVersion = '1.1'
+    ) {
+        return new Request($method, $uri, $headers, $body, $protocolVersion);
+    }
+
+    /**
+     * Get a request using a simplified array of options.
+     *
+     * @param  null|string $method
+     * @param  null|string $uri
+     * @param  array $options
+     * @return Request
+     */
+    public function getRequestWithOptions($method, $uri, array $options = [])
+    {
+        $func    = new \ReflectionMethod($this, 'getRequest');
```

Comment:
[Using array merge instead, which is not as DRY but doesn't use reflection.](https://github.com/shadowhand/oauth2-client/pull/3/files#r32169493)